index.js

```diff
@@ -1,59 +1,108 @@
+     * @param  {Object}         config.ecosystem            Object with ecosystem values
+     * @param  {Array}          config.executor             Array of executors to load
+     * @param  {Object}         config.redisConnection      Connection details for redis
      * @param  {String}         config.executor[x].name     Name of the executor NPM module to load
      * @param  {String}         config.executor[x].options  Configuration to construct the module with
      */
     constructor(config = {}) {
-        const ecosystem = config.ecosystem || {};
         const executorConfig = config.executor;
-        const defaultPlugin = config.defaultPlugin;
+        const ecosystem = config.ecosystem || {};
 
         if (!executorConfig || !(Array.isArray(executorConfig)) || executorConfig.length === 0) {
             throw new Error('No executor config passed in.');
         }
 
         super();
 
-        executorConfig.forEach((plugin) => {
-            let ExecutorPlugin;
+        this.ecosystem = ecosystem;
+        this.defaultPlugin = config.defaultPlugin;
 
-            try {
-                // eslint-disable-next-line global-require, import/no-dynamic-require
-                ExecutorPlugin = require(`screwdriver-executor-${plugin.name}`);
-            } catch (err) {
-                console.error(err);
+        this.executorConfigs = {};
+        executorConfig.forEach((plugin) => {
+            this.executorConfigs[plugin.name] = Object.assign({ ecosystem }, plugin.options);
 
-                return;
+            if (!this.defaultPlugin) {
+                this.defaultPlugin = plugin.name;
             }
+        });
+
+        // eslint-disable-next-line new-cap
+        this.queue = new Resque.queue({ connection: config.redisConnection });
 
-            const options = Object.assign({ ecosystem }, plugin.options); // Add ecosystem to executor options
-            const executorPlugin = new ExecutorPlugin(options);
+        this.connected = false;
+        this.queue.connect(() => {
+            this.connected = true;
+        });
+    }
 
-            if (defaultPlugin && defaultPlugin === plugin.name) {
-                // set the default plugin to the desired one
-                this.defaultExecutor = executorPlugin;
-            } else if (!defaultPlugin && !this.defaultExecutor) {
-                // When given no default plugin, use the first module we instantiated
-                this.defaultExecutor = executorPlugin;
+    /**
+     * Get the full config for a job
+     * @method getFullConfig
+     * @param {Object} buildConfig  build configuration
+     * @return {Object}
+     */
+    getFullConfig(buildConfig) {
+        const annotations = buildConfig.annotations || {};
+        const executorType = getPluginName(
+            annotations[ANNOTATION_EXECUTOR_TYPE] || this.defaultPlugin);
+        const options = {
+            buildConfig,
+            executor: {}
```

Comment:
Currently the workers are starting a new executor each time, and getting their config through the queue since they don't otherwise have access to the API, which is where the executor config is stored.